### ISSUE TITLE
Nuxt 3.7.0 - devServer https config not taken into account

### ISSUE THREAD

**Issue by JulienVanelian** [nuxt/nuxt#22846]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.20.0`
- Nuxt Version:     `3.7.0`
- CLI Version:      `3.7.0`
- Nitro Version:    `2.6.1`
- Package Manager:  `yarn@1.22.19`
- Builder:          `-`
- User Config:      `modules`, `experimental`, `i18n`, `ignore`, `components`, `css`, `colorMode`, `nitro`, `app`, `devServer`, `devtools`
- Runtime Modules:  `@vueuse/nuxt@10.4.0`, `@unocss/nuxt@0.55.3`, `@pinia/nuxt@0.4.11`, `@nuxtjs/color-mode@3.3.0`, `@nuxtjs/i18n@8.0.0-rc.3`
- Build Modules:    `-`

### Reproduction

For obvious reasons I can't share the certs

### Describe the bug

In the nuxt.config.ts, setting the certs for https devServer doesn't have any effect. It was working on 3.6.5.

```
export default defineNuxtConfig({
  devServer: {
    https: {
      key: './server.key',
      cert: './server.crt'
    }
  }
})
```

### Logs

```shell-script
Nuxt starts printing the following, completely ignoring the https devServer config:

yarn run v1.22.19
$ nuxi dev
Nuxt 3.7.0 with Nitro 2.6.1                                          11:01:54 PM
                                                                     11:01:54 PM
  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

ℹ Vite client warmed up in 1137ms                                   11:01:57 PM
✔ Nitro built in 723 ms
```

**Comment by mario-neuhold** [comment]
same here. thought it kinda gets taken into account. if you start it with `dev --https` it uses thouse key files from the config.

**Comment by george-martinec** [comment]
same here

**Comment by pi0** [comment]
Sorry for the inconvenience. Several fixes were added to [nuxi 3.7.2](https://github.com/nuxt/cli/releases/tag/v3.7.2) in order to address this issue.

You can self-update using `npx nuxi upgrade --force` and make sure to use 3.7.2+ using `npx nuxi info`. 

If you still having issues, don't hesitate to ping me.

**Comment by DamianGlowala** [comment]
@JulienVanelian, did the nuxi upgrade fix it for you? Are you now able to run the dev server with HTTPS? Sadly, I can't get it to work having run `npx nuxi upgrade --force`.

```
- Operating System: Windows_NT
- Node Version:     v16.20.1
- Nuxt Version:     3.7.1-28224571.2d469711
- CLI Version:      3.7.4-1693421082.011a60f
- Nitro Version:    2.6.3-28223518.83e980b
- Package Manager:  pnpm@8.6.7
```

**Comment by mario-neuhold** [comment]
@DamianGlowala since it's a nuxt-cli problem i continued in an existing issue there: https://github.com/nuxt/cli/issues/99
but as mentioned last week, you can get it to run properly just by adding --https in the dev script of package.json: `"dev": "nuxt dev --https"`.

**Comment by mario-neuhold** [comment]
fixed with nuxi 3.8.0 https://github.com/nuxt/cli/pull/146
